lib/trino-hive-formats/src/main/java/io/trino/hive/formats/esri/EsriDeserializer.java

```
@@ -0,0 +1,371 @@
+            }
+        }
+    }
+
+    private static DecodedTimestamp createDecodedTimestamp(Timestamp timestamp)
+    {
+        long millis = timestamp.getTime();
+        long epochSeconds = floorDiv(millis, (long) MILLISECONDS_PER_SECOND);
+        long fractionalSecond = floorMod(millis, (long) MILLISECONDS_PER_SECOND);
+        int nanosOfSecond = toIntExact(fractionalSecond * (long) NANOSECONDS_PER_MILLISECOND);
+
+        return new DecodedTimestamp(epochSeconds, nanosOfSecond);
+    }
+
+    private static void serializeDecimal(String value, DecimalType decimalType, BlockBuilder builder)
+    {
+        BigDecimal bigDecimal;
+        try {
+            bigDecimal = new BigDecimal(value).setScale(DecimalConversions.intScale(decimalType.getScale()), HALF_UP);
+        }
+        catch (NumberFormatException e) {
+            throw new NumberFormatException(format("Cannot convert '%s' to %s. Value is not a number.", value, decimalType));
+        }
+
+        if (overflows(bigDecimal, decimalType.getPrecision())) {
+            throw new IllegalArgumentException(format("Cannot convert '%s' to %s. Value too large.", value, decimalType));
+        }
+
+        if (decimalType.isShort()) {
+            decimalType.writeLong(builder, bigDecimal.unscaledValue().longValueExact());
+        }
+        else {
+            decimalType.writeObject(builder, Int128.valueOf(bigDecimal.unscaledValue()));
+        }
+    }
+
+    private Date parseDate(JsonParser parser)
+            throws IOException
+    {
+        Date jsd = null;
+        if (JsonToken.VALUE_NUMBER_INT.equals(parser.getCurrentToken())) {
+            long epoch = parser.getLongValue();
+            jsd = new Date(epoch - (long) tz.getOffset(epoch));
+        }
+        else {
+            try {
+                long epoch = parseTime(parser.getText(), "yyyy-MM-dd");
+                jsd = new Date(epoch + 43200000L);
+            }
+            catch (ParseException e) {
```

Returning `null` here is probably something we want to make a pluggable behavior in the future, and maybe initially we probably want to prefer to fail loudly here instead of preserving the existing Hive behavior (see: https://github.com/trinodb/trino/issues/23673)

---

Updated to throw exception when no format matches